src/api/entities/SecurityToken/Checkpoints.ts

```diff
@@ -24,4 +26,38 @@ export class Checkpoints extends Namespace<SecurityToken> {
    *   - Security Token Owner
    */
   public create: ProcedureMethod<void, SecurityToken>;
+
+  /**
+   * Retrieve all Checkpoints created on this Security Token, together with their corresponding creation Date
+   *
+   * @note supports pagination
+   */
+  public async get(
+    paginationOpts?: PaginationOptions
+  ): Promise<ResultSet<CheckpointWithCreationDate>> {
+    const {
+      parent: { ticker },
+      context,
+    } = this;
+
+    const rawTicker = stringToTicker(ticker, context);
+
+    const { entries, lastKey: next } = await requestPaginated(
+      context.polymeshApi.query.checkpoint.timestamps,
+      { paginationOpts, arg: rawTicker }
+    );
+
+    const now = new Date();
+    const data = entries
+      .map(([{ args: [, id] }, timestamp]) => ({
+        checkpoint: new Checkpoint({ id: u64ToBigNumber(id), ticker }, context),
+        createdAt: momentToDate(timestamp),
+      }))
+      .filter(({ createdAt }) => createdAt <= now);
```

Comment:
Why do you need these filter?

Reply:
The query also returns the next scheduled checkpoint's time